app/views/questionnaire.py

```diff
@@ -326,12 +333,14 @@ def post_block_handler(
     return redirect(next_location.url())
 
 
-def perform_list_action(
+def perform_list_action(  # noqa: C901 pylint: disable=too-complex
```

Comment:
Can this not be refactored rather than disabling lint warnings?

Reply:
Ideally yes. I'm still looking at how best to do that.